### DISCUSSION TITLE
[Enhancement]: Cleaning up old backups from remote S3 storage

### DISCUSSION THREAD

**Discussion by marcinlawnik** [coollabsio/coolify#4368] [Improvement Requests]
### Request Type

Improvement

### Description

I use S3-compatible storage from a provider other than AWS. It does not have a lifecycle/retention policy feature, which leads to it filling up quickly with backups. Unlike AWS, it also does not bill by operations, which means regularly reading and deleting stuff has no additional cost.

Other backup solutions I use (for example laravel-backup from spatie) have a "retention" feature they take care of themselves  - i.e. after a successful backup they delete backups older than specified N days from remote storage.

Here's a page from their docs (I actually see they have more complicated/opinionated rules, which seem fine).
https://spatie.be/docs/laravel-backup/v8/cleaning-up-old-backups/overview

This could be an option on the backup screen for a resource.

**Comment by flobamediaacc** [comment]
Good suggestion! It would be great if you could set how long the remote backups are kept.

**Comment by peaklabs-dev** [comment]
@marcinlawnik and @flobamediaacc This has now been integrated, I added this feature in a recent release.
Preview:

![image](https://github.com/user-attachments/assets/997348e0-da13-400e-873d-7cf9a903e8f7)

**Reply by marcinlawnik** [reply]
@peaklabs-dev awesome ;)
